### DISCUSSION TITLE
when I finish change the source code, then how can i start to use it?

### DISCUSSION THREAD

**Discussion by qigeqigeqi** [coder/code-server#4001] [Q&A] [answered]
i want to annotate some function, then i change the source code, how can i use the changed code-server? Thx!!!!!

**Comment by lg-nineoneninethreenine** [comment]
There might be a way to build code-server from yarn, but it's unlikely, as they have strange methods of building it. but i think @jsjoeio knows.

**Reply by qigeqigeqi** [reply]
now i use the install.sh in the code which i have changed

**Reply by lg-nineoneninethreenine** [reply]
You changed the `install.sh`? Is it on a GitHub repo?

**Reply by qigeqigeqi** [reply]
no, i change th lib/vs/..... then i run install.sh, i try to change the function of code-server when the install.sh finish, is it ok?

**Reply by lg-nineoneninethreenine** [reply]
Put it on a GitHub repo or fork **this** one, and apply the changes. Then, type this on Terminal:
```bash
curl -fsSL https://github.com/lyq1198853167/code-server/blob/main/install.sh | sh
```
and check if that works.

**Reply by qigeqigeqi** [reply]
in ci/build/ , there is some build script, i want to build a packe but when i run ./build-packages.sh there is some erro:

tar: ./release-standalone: Cannot stat: No such fiile or directory
tar: Exiting with failure status due to previous errors 

but when i create , it means useless

**Reply by lg-nineoneninethreenine** [reply]
> in ci/build/ , there is some build script, i want to build a packe but when i run ./build-packages.sh there is some erro:
> 
> tar: ./release-standalone: Cannot stat: No such fiile or directory
> tar: Exiting with failure status due to previous errors
> 
> but when i create , it means useless

Well, this can be answered by @jsjoeio or @code-asher.

**Comment by lg-nineoneninethreenine** [comment]
Referring https://github.com/cdr/code-server/discussions/4001#discussioncomment-1193396

Answer might be:

https://github.com/cdr/code-server/discussions/4001#discussioncomment-1193481

**Comment by code-asher** [comment, marked as answer]
If you are editing a build/distribution of code-server you can just restart code-server.

But if you are editing the repository source then you'll need to build a new code-server from it. You can find information on how to do this here: https://github.com/cdr/code-server/blob/main/docs/CONTRIBUTING.md#development-workflow
